### ISSUE TITLE
youtube player & iframe issues _ Not receiving keyboard focus

### ISSUE THREAD

**Issue by HerinHentry** [mr-bobz/talarts#42]
# iframe width="960" height="570" class="youtube-player" src="http://www.youtube.com/embed/BWk_q3Y6IvI?version=3&amp;rel=1&amp;fs=1&amp;showsearch=0&amp;showinfo=1&amp;iv_load_policy=1&amp;wmode=transparent" frameborder="0" type="text/html">

**Comment by HerinHentry** [comment]
Format:
<...iframe src="banner-ad.html" id="testiframe" 
  name="testiframe" title="Advertisement">

**Comment by mr-bobz** [comment]
YouTube removes the iframe title some times, added an aria-label also as backup

Also replaced SiteOrigin embed with direct iframe and tweaked youtube params

**Comment by HerinHentry** [comment]
<...iframe width="960" height="570" class="youtube-player" src="http://www.youtube.com/embed/FprQKc0bCKY?fs=1&amp;showsearch=0&amp;showinfo=1&amp;iv_load_policy=1&amp;wmode=transparent&amp;controls=2&amp;disablekb=1&amp;modestbranding=1&amp;rel=0" frameborder="0" aria-label="Promotional Video">TALA promotional video by Pitch...</iframe>

@christosptr Can you please advise on this? Title has been removed by youtube. Aria-label will this be enough to meet a11y requirement?

**Comment by HerinHentry** [comment]
@mr-bobz : The youtube player controls like share, cc, play and every control was getting keyboard focus when it was a site origin embed. But, now the youtube does not receive keyboard focus other than play. Also, the keyboard focus on the play is not visible. I tested this with Chrome, Firefox, IE. Same issue.

**Comment by mr-bobz** [comment]
Should be fixed now, pls retest, thanks

**Comment by HerinHentry** [comment]
It's working now. This is the speech view of the NVDA. Are we able to tweak **Button unavailable & blank?**

![image](https://cloud.githubusercontent.com/assets/15111528/10991190/6c15bd6e-84af-11e5-9e2f-9dd611523921.png)

**Comment by HerinHentry** [comment]
@mr-bobz Please see my previous comment. As well as I found that the youtube player is receiving focus in IE & firefox, Not on Chrome. What could be the reason for this?

**Comment by christosptr** [comment]
Hi Herin i find that focus on chrome is received on the play button in the
middle of the video then you can play the video. Once you start playing the
video the rest of the controls are shown. but the focus is set at the end
of the video
so you need to press shift + tab to get to the video player. When you press
the shift tab and the focus is set on the video player you first here the
menu been announced first

Not sure why this is the case. That is hear the Tala menu

Cheers Christos
Send from my mobile device

**Comment by mr-bobz** [comment]
We cannot change/tweak/override contents in an iframe, especially when loaded from an external url. Only other option is to find an accessible video player plugin for wordpress

**Comment by HerinHentry** [comment]
*_34. Video (20 points).  *_
For video with soundtrack, provide synchronized captions.  Provide an html text description for video without sound.  In addition, provide synchronized audio description if the video cannot by understood from the soundtrack alone. If not synchronized, audio description must be provided as text with the video link. Each instance of inadequate accommodation is an error.  
This does not specify about the video player accessibility. @christosptr , @mr-bobz : If we agree to the way it works, we can close this issue. As we have lot of other issues to concentrate on (menu , donation), we can agree on having this video. 
@christosptr : Should accessibility page have information on how to play video on Chrome?

**Comment by christosptr** [comment]
I think it fine the way it is. Closing this issue
